index.bs

```diff
@@ -1872,6 +1841,312 @@ partial interface MLGraphBuilder {
     </div>
 </div>
 
+### The lstm() method ### {#api-mlgraphbuilder-lstm}
+Long Short-Term Memory [[LSTM]] recurrent network uses an input, output, forget, and cell gate to compute the output state that rolls into the output across the temporal sequence of the network.
+<script type=idl>
+enum MLLstmWeightLayout {
+  "iofg", // input-output-forget-cell gate ordering
+  "ifgo"  // input-forget-cell-output gate ordering
+};
+
+dictionary MLLstmOptions {
+  MLOperand bias;
+  MLOperand recurrentBias;
+  MLOperand peepholeWeight;
+  MLOperand initialHiddenState;
+  MLOperand initialCellState;
+  boolean returnSequence = false;
+  MLRecurrentNetworkDirection direction = "forward";
+  MLLstmWeightLayout layout = "iofg";
+  sequence<MLActivation> activations;
+};
+
+partial interface MLGraphBuilder {
+  sequence<MLOperand> lstm(MLOperand input, MLOperand weight, MLOperand recurrentWeight, 
+                           unsigned long steps, unsigned long hiddenSize, 
+                           optional MLLstmOptions options = {});
+};
+</script>
+<div algorithm=lstm>
+    **Arguments:**
+        - *input*: an {{MLOperand}}. The input 3-D tensor of shape [steps, batch_size, input_size].
+        - *weight*: an {{MLOperand}}. The 3-D input weight tensor of shape [num_directions, 4 * hidden_size, input_size]. The ordering of the weight vectors in the second dimension of the tensor shape is specified according to the *options.layout* argument.
```

Comment:
Naming inconsistencies:
`hidden_size` (line 1873) vs `hiddenSize` (line 1866)
`num_directions` vs `numDirections (line 1893)

Reply:
Thanks for pointing this out. But I'm not going to address this issue in this PR. It probably needs a round of cleanup.